### PR TITLE
PS performance cleanups

### PR DESCRIPTION
**Pull request by baude**
By not getting data that we don't need and adding more locking
functions under the batch, we are able to cut the time for listing
50 containers in half.  More to come.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: cmd/podman/ps.go [hunk 1]
```diff
@@ -418,35 +421,49 @@ func getTemplateOutput(containers []*libpod.Container, opts psOptions) ([]psTemp
...
+			if !opts.size && !opts.namespace {
+				return nil
+			}
+
+			if opts.namespace {
+				pid, err = c.PID()
+				if err != nil {
+					return errors.Wrapf(err, "unable to obtain container pid")
+				}
+				ns = getNamespaces(pid)
```
Comment (mheon) [inline cmd/podman/ps.go]: I think you can move this after the `Batch()` inside another `if opts.namespace` - it doesn't call into the container at all, I think.

Comment (baude) [reply cmd/podman/ps.go]: fixed

File: cmd/podman/ps.go [hunk 2]
```diff
@@ -458,15 +474,14 @@ func getTemplateOutput(containers []*libpod.Container, opts psOptions) ([]psTemp
 			logrus.Errorf("couldn't get some ps information, error getting artifact %q: %v", ctr.ID(), err)
 		}
 
+		ns = getNamespaces(pid)
```
Comment (mheon) [inline cmd/podman/ps.go]: Does this blow up if pid == 0 because we never got it? Maybe put in if opts.namespace?

**Comment by mheon** [comment]
LGTM, merging once tests green

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**